meshnet/server.go

```diff
@@ -1254,16 +1255,126 @@ func (s *Server) RemovePeer(
 	}, nil
 }
 
+func (s *Server) RenamePeer(
+	ctx context.Context,
+	req *pb.RenamePeerRequest,
+) (*pb.RenamePeerResponse, error) {
+	if !s.ac.IsLoggedIn() {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_NOT_LOGGED_IN,
+			},
+		}, nil
+	}
+
+	var cfg config.Config
+	if err := s.cm.Load(&cfg); err != nil {
+		s.pub.Publish(err)
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_CONFIG_FAILURE,
+			},
+		}, nil
+	}
+
+	// check if meshnet is enabled
+	if !cfg.Mesh {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_MeshnetErrorCode{
+				MeshnetErrorCode: pb.MeshnetErrorCode_NOT_ENABLED,
+			},
+		}, nil
+	}
+
+	token := cfg.TokensData[cfg.AutoConnectData.ID].Token
+
+	// check info and re-register if needed
+	if !s.mc.IsRegistrationInfoCorrect() {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_CONFIG_FAILURE,
+			},
+		}, nil
+	}
+
+	// TODO: sometimes IsRegistrationInfoCorrect() re-registers the device => cfg.MeshDevice.ID can be different.
+	resp, err := s.reg.List(token, cfg.MeshDevice.ID)
+
+	if err != nil {
+		if errors.Is(err, core.ErrUnauthorized) {
+			// TODO: check what happens with cfg.Mesh
+			if err := s.cm.SaveWith(auth.Logout(cfg.AutoConnectData.ID)); err != nil {
+				s.pub.Publish(err)
+				return &pb.RenamePeerResponse{
+					Response: &pb.RenamePeerResponse_ServiceErrorCode{
+						ServiceErrorCode: pb.ServiceErrorCode_CONFIG_FAILURE,
+					},
+				}, nil
+			}
+			return &pb.RenamePeerResponse{
+				Response: &pb.RenamePeerResponse_ServiceErrorCode{
+					ServiceErrorCode: pb.ServiceErrorCode_NOT_LOGGED_IN,
+				},
+			}, nil
+		}
+
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_ServiceErrorCode{
+				ServiceErrorCode: pb.ServiceErrorCode_API_FAILURE,
+			},
+		}, nil
+	}
+
+	peer := s.getPeerWithId(req.GetIdentifier(), resp)
+
+	if peer == nil {
+		return &pb.RenamePeerResponse{
+			Response: &pb.RenamePeerResponse_UpdatePeerErrorCode{
+				UpdatePeerErrorCode: pb.UpdatePeerErrorCode_PEER_NOT_FOUND,
+			},
+		}, nil
+	}
+
+	if req.Nickname == "" && peer.Nickname == "" {
```

Comment:
What is this situation? When can it happen? I why we give empty response to user?

Reply:
The user tries to remove the nickname from a peer that already doesn't have nickname.

Reply:
updated and in this case an error will be returned if peer doesn't have a nickname.